core/src/query/store.rs

```diff
@@ -197,13 +199,25 @@ impl LiveQueryStoreHandle {
         self.construct_query_response(query_id, cursor.cursor.map(NonZeroU64::get), live_query)
     }
 
-    fn insert(&self, query_id: String, live_query: LiveQuery) -> Result<()> {
+    /// Remove query from the storage if there is any.
+    ///
+    /// Returns `true` if query was removed, `false` otherwise.
+    ///
+    /// # Errors
+    ///
+    /// - Returns [`Error::ConnectionClosed`] if [`QueryService`] is dropped,
+    /// - Otherwise throws up query output handling errors.
+    pub fn forget_query(&self, query_id: QueryId) -> Result<bool> {
```

Comment:
```suggestion
    pub fn remove_query(&self, query_id: QueryId) -> Result<bool> {
```

`forget` to me implies a memory leak